### DISCUSSION TITLE
Non-apple code signed bundles

### DISCUSSION THREAD

**Discussion by Moosems** [dennisvang/notsotuf-example#27] [Q&A] [answered]
On Mac you need your application to be code-signed by a certificate used by apple. If I don't have one of these certificates and people use the update functinality, will it require them to manually authorize it every update?

**Comment by dennisvang** [comment, marked as answer]
@Moosems Unfortunately I'm not familiar with the specifics of Apple's code signing requirements, so I cannot answer that. 

Perhaps @JessicaTegner, @cbenhagen, or another mac user knows the answer?

Otherwise I think the best option would be to try it on a clean mac with a minimal app (e.g. based on the example app).

On Windows I've not seen any warnings during updates, after approving an unsigned app upon first installation. However, that may also depend on windows defender's heuristics.

As far as tufup is concerned, ***if*** you do any code-signing, it needs to be done ***before*** adding the bundle to the update repository, as illustrated by  https://github.com/dennisvang/tufup/issues/35.
